### PR TITLE
ToastyBoi 2.0 

### LINKED ISSUE #322: Missing toasts for when visibility is toggled, map missing on publish story toast
# Description 
The publish toast should have a map icon to the left of it like in Figma. 
image
When multi-select is toggled, there should be a toast after the "hide" or "show" buttons are clicked.
https://www.figma.com/file/uVqnweL2ApqvPHWzQYmZ1t/Dashboard?node-id=1159%3A2297 

# Potential Solution
- recycle the toast component from publish for the visibility buttons
- add the map icon from Material UI icons.

### PR DESCRIPTION
**Pull request by liu-jeffrey**
Closes https://github.com/uwblueprint/shoe-project/issues/322

Showing / hiding multiple stories now has their own toasts. Toasts now have icons, and you can have consecutive toasting :D

### CODE REVIEW

**Review by nimwijetunga — APPROVED** [review]
Looks good to me! But just wait for @hanlinc27 and @Shehryar21 before merging

File: ui/src/pages/Admin/AllStories/index.tsx [hunk 1]
```diff
@@ -533,8 +552,8 @@ export const AllStories: React.FC = () => {
 
   const toast = React.useRef(null);
 
-  const showToast = (message: string) => {
-    toast.current.showToast(message);
+  const showToast = (message: string, icon: any) => {
```
Comment (nimwijetunga) [inline ui/src/pages/Admin/AllStories/index.tsx]: can we address this warning for the type for icon?

Comment (liu-jeffrey) [reply ui/src/pages/Admin/AllStories/index.tsx]: wait which warning?

Comment (hanlinc27) [reply ui/src/pages/Admin/AllStories/index.tsx]: Material icons are just svg components so let's type it as `React.ComponentType` ?

Comment (nimwijetunga) [reply ui/src/pages/Admin/AllStories/index.tsx]: image

File: ui/src/pages/Admin/AllStories/index.tsx [hunk 2]
```diff
@@ -456,6 +463,12 @@ export const AllStories: React.FC = () => {
         } as React.ChangeEvent<HTMLInputElement>;
         handleSwitchChangeCheckbox(changeEvent, getStoryByID(s));
       });
+      showToast(
+        numVisibleChecked.toString() +
+          (numVisibleChecked > 1 ? " stories" : " story") +
+          " hidden from map.",
+        VisibilityOffOutlinedIcon
```
Comment (hanlinc27) [inline ui/src/pages/Admin/AllStories/index.tsx]:
```suggestion
 `${numVisibleChecked.toString()}${
          numVisibleChecked > 1 ? " stories" : " story"
        } hidden from map.`
```

File: ui/src/pages/Admin/AllStories/index.tsx [hunk 3]
```diff
@@ -466,6 +479,12 @@ export const AllStories: React.FC = () => {
         } as React.ChangeEvent<HTMLInputElement>;
         handleSwitchChangeCheckbox(changeEvent, getStoryByID(s));
       });
+      showToast(
+        numHiddenChecked.toString() +
```
Comment (hanlinc27) [inline ui/src/pages/Admin/AllStories/index.tsx]: preferably same as above

File: ui/src/components/ToastyBoi.tsx [hunk 4]
```diff
@@ -45,19 +72,31 @@ const ToastyBoi = forwardRef((props, ref) => {
...
   const handleClose = () => {
-    setToastState({
-      ...toastState,
-      open: false,
-    });
+    setOpen(false);
   };
 
+  const handleExited = () => {
+    setMessageInfo(undefined);
```
Comment (hanlinc27) [inline ui/src/components/ToastyBoi.tsx]: why are all of these undefined instead of setting it as empty string?

**Review by hanlinc27 — APPROVED** [review]
Toast to this 🥂 Looks great on my side :)) 
giphy (2)

**Review by Shehryar21 — APPROVED** [review]
looks good!!!

**Comment by nimwijetunga** [comment]
@hanlinc27 I addressed your comments can you take a quick look before i merge this?

**Pull request merged**